### PR TITLE
add koji_parent plugin

### PR DESCRIPTION
**Pull request by lcarva**
Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

File: atomic_reactor/plugins/pre_koji_parent.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        self.koji_session = create_koji_session(koji_hub, koji_auth_info)
+
+        self.poll_interval = poll_interval
+        self.poll_timeout = poll_timeout
+
+        self._parent_image_nvr = None
+        self._parent_image_build = None
+        self._poll_start = None
+
+        # TODO: Handle scratch builds?
```
Comment (twaugh) [inline atomic_reactor/plugins/pre_koji_parent.py]: I think let's not run this plugin for scratch builds.

File: atomic_reactor/plugins/pre_koji_parent.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        :param koji_ssl_certs_dir: str, path to "cert", "ca", and "serverca"
+                                   used when Koji's identity certificate is not trusted
+        :param poll_interval: int, seconds between polling for Koji build
+        :param poll_timeout: int, max amount of seconds to wait for Koji build
+        """
+        super(KojiParentPlugin, self).__init__(tasker, workflow)
+
+        koji_auth_info = None
+        if koji_ssl_certs_dir:
+            koji_auth_info = {
```
Comment (twaugh) [inline atomic_reactor/plugins/pre_koji_parent.py]: Do we want coverage here?

Comment (lcarva) [inline atomic_reactor/plugins/pre_koji_parent.py]: Added.

File: atomic_reactor/plugins/pre_koji_parent.py (hunk 1 above)
Comment (lcarva) [inline atomic_reactor/plugins/pre_koji_parent.py]: Removed TODO comment.

File: atomic_reactor/plugins/pre_koji_parent.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        while self.is_within_timeout():
+            if self.has_parent_image_build():
+                self.log.info('Parent image Koji build found')
+                break
+            time.sleep(self.poll_interval)
+
+    def start_polling_timer(self):
+        self._poll_start = time.time()
+
+    def is_within_timeout(self):
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_koji_parent.py]: Maybe this should include `start_polling_timer` too:
```
if not self._poll_start:
    self._poll_start = time.time()
return (time.time() - self._poll_start) < self.poll_timeout
```

Comment (lcarva) [inline atomic_reactor/plugins/pre_koji_parent.py]: Although they work together, they are separate things. I'd rather keep them separate.

Comment (vrutkovs) [inline atomic_reactor/plugins/pre_koji_parent.py]: Okay, agreed

**Comment by lcarva** [comment]
Release notes updated.

**Pull request merged**